pyanaconda/install_tasks.py

```diff
@@ -0,0 +1,424 @@
+# installation_tasks.py
+# Container classes for running of installation tasks.
+#
+# Copyright (C) 2016  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from threading import RLock
+from pyanaconda.isignal import Signal
+import time
+import logging
+log = logging.getLogger("anaconda")
+
+
+class TaskQueue(list):
+    """TaskQueue represents a queue of TaskQueues or Tasks.
+
+    TaskQueues and Tasks can be mixed in a single TaskQueue.
+    """
+
+    def __init__(self, name, status_message=None):
+        super(TaskQueue, self).__init__()
+        self._name = name
+        self._status_message = status_message
+        self._lock = RLock()
+        self._current_task_number = None
+        self._current_queue_number = None
+        self._running = False
+        self._done = False
+        self._parent = None
```

Comment:
Nope, running and done should never be True at the same time. This is how it should go (running, done):
False, False - not yet started
True, False - once start has been called & as long as processing is ongoing
False, True - once finished

Switching between these states should be (hopefully ;-) ) atomic. :)